web_app_client.py

```diff
@@ -0,0 +1,173 @@
+import time
+
+import requests
+import json
+from datetime import datetime
+from dataclasses import dataclass
+from typing import Optional, Dict, Union
+
+
+@dataclass
+class Status:
+    """
+    Represents the status of an uploaded presentation.
+
+    Attributes:
+        status (str): The status of the upload.
+        filename (str): The original filename of the presentation.
+        timestamp (datetime): The timestamp of the upload.
+        explanation (Optional[str]): The explanation of the upload, if available.
+    """
+
+    status: str
+    filename: str
+    timestamp: datetime
+    explanation: Optional[str]
+
+    def is_done(self) -> bool:
+        """
+        Checks if the status is 'done'.
+
+        Returns:
+            bool: True if the status is 'done', False otherwise.
+        """
+        return self.status == 'done'
+
+
+class WebAppClient:
+    """
+    Client for interacting with the Web API.
+
+    Args:
+        base_url (str): The base URL of the Web API.
+
+    Attributes:
+        base_url (str): The base URL of the Web API.
+    """
+
+    def __init__(self, base_url: str):
+        """
+        Initializes the WebAppClient with the base URL of the Web API.
+
+        Args:
+            base_url (str): The base URL of the Web API.
+        """
+        self.base_url = base_url
+
+    def upload(self, file_path: str, email: str = None) -> str:
```

Comment:
Separate code to small functions.  Read more: https://softwareengineering.stackexchange.com/questions/308108/when-is-it-appropriate-to-make-a-separate-function-when-there-will-only-ever-be